### PR TITLE
feat(v8): Update node docs for v8

### PR DESCRIPTION
**Pull request by mydea**
This is a draft until v8 is released!

### CODE REVIEW

File: docs/platforms/javascript/common/performance/database/index.mdx [hunk 1]
```diff
@@ -17,19 +18,16 @@ Node.js integrations support tracking database queries as spans.
...
-- `pg-native` (Postgres)
-- `mongodb` (Mongo)
-- `mongoose` (Mongo)
-- `mysql` (MySQL)
-- `graphql` (GraphQL)
-- `apollo-server-core` (Apollo)
-- `@nestjs/graphql` (Apollo)
+- `graphql` (including Apollo Server)
+- `mongo`
+- `mongoose`
```
Comment (s1gr1d) [inline docs/platforms/javascript/common/performance/database/index.mdx:26]: and also 
`- redis (ioredis)`

File: includes/node-esm-compatibility.mdx [hunk 2]
```diff
@@ -0,0 +1,5 @@
+## Using Sentry SDK with ESM
+
+Instrumentation works out of the box for CommonJS (CJS) applications based on `require()` calls. This means that as long as your application is either natively in CJS, or compiled at build time to CJS (which is the case for most TypeScript apps, for example), everything will work without any further setup.
+
+Because of compatibility issues, today the Sentry SDK does not support ESM (based on `import`). We are working on a solution for this, and hope to support this soon.
```
Comment (s1gr1d) [inline includes/node-esm-compatibility.mdx:5]:
```suggestion
Due to compatibility issues, the Sentry SDK does not currently support ESM (based on `import`). We are working on a solution for this and hope to support this soon.
```

**Review by s1gr1d — APPROVED** [review]
Thanks for updating 🙌🏻

**Comment by timfish** [comment]
I think the `Configure metrics` section can now be removed from all SDKs since they're automatically configures when you first use the functions.

For example, there's no need to do this anymore:
https://sentry-docs-git-v8-docs.sentry.dev/platforms/javascript/guides/node/metrics/

File: docs/platforms/javascript/guides/connect/index.mdx [hunk 3]
```diff
@@ -6,7 +6,19 @@ categories:
   - server
 ---
 
-Add `@sentry/node` as a dependency:
+<PlatformContent includePath="getting-started-primer" />
+
+This guide explains how to setup Sentry in your Connect application.
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/connect/index.mdx:11]:
```suggestion
This guide explains how to set up Sentry in your Connect application.
```

File: docs/platforms/javascript/guides/connect/performance/index.mdx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+## Automatic Instrumentation
+
+See <PlatformLink to="/performance/instrumentation/automatic-instrumentation">Automatic Instrumentation</PlatformLink> to learn about all the things that the SDK automatically instruments for you.
+
+## Custom Instrumentation
+
+You can also manually start spans to instrument specific parts of your code. This is useful when you want to measure the performance of a specific operation or function.
+
+See <PlatformLink to="/performance/instrumentation/custom-instrumentation">Custom Instrumentation</PlatformLink> on how to manually start spans.
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/connect/performance/index.mdx:24]:
```suggestion
See <PlatformLink to="/performance/instrumentation/custom-instrumentation">Custom Instrumentation</PlatformLink> to learn how to manually start spans.
```

File: docs/platforms/javascript/guides/connect/performance/index.mdx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+If you’re adopting Performance in a high-throughput environment, we recommend testing prior to deployment to ensure that your service’s performance characteristics maintain expectations.
+
+</Note>
+
+<PlatformContent includePath="getting-started-config" />
+
+## Automatic Instrumentation
+
+See <PlatformLink to="/performance/instrumentation/automatic-instrumentation">Automatic Instrumentation</PlatformLink> to learn about all the things that the SDK automatically instruments for you.
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/connect/performance/index.mdx:18]:
```suggestion
See <PlatformLink to="/performance/instrumentation/automatic-instrumentation">Automatic Instrumentation</PlatformLink> to learn about everything the SDK automatically instruments for you.
```

File: docs/platforms/javascript/guides/connect/performance/index.mdx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+---
+title: Performance Monitoring
+description: "Learn more about how to configure our Performance integrations to get the best experience out of it."
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/connect/performance/index.mdx:3]:
```suggestion
description: "Learn more about how to configure our Performance integrations to get the best experience."
```

File: docs/platforms/javascript/guides/gcp-functions/index.mdx [hunk 7]
```diff
@@ -6,53 +6,61 @@ categories:
   - serverless
 ---
 
-Add `@sentry/serverless` as a dependency to `package.json`:
+<Note>
+  This guide is for Version 8.0.0 and up of `@sentry/google-cloud-serverless`.
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/gcp-functions/index.mdx:10]:
```suggestion
  This guide is for version 8.0.0 and up of `@sentry/google-cloud-serverless`.
```

File: docs/platforms/javascript/guides/gcp-functions/performance/instrumentation/automatic-instrumentation.mdx [hunk 8]
```diff
@@ -4,39 +4,28 @@ description: "Learn what transactions are captured after tracing is enabled."
...
+- Fastify routes & middlewares
+- HTTP requests made with `http`, `https` and `fetch`
+- DB queries made with:
+- `mysql`
+- `mysql2`
+- `pg`
+- `graphql` (including Apollo Server)
+- `mongo`
+- `mongoose`
+- `prisma`
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/gcp-functions/performance/instrumentation/automatic-instrumentation.mdx:18]:
```suggestion
```

File: docs/platforms/javascript/guides/koa/index.mdx [hunk 9]
```diff
@@ -6,157 +6,54 @@ categories:
...
 
-Add `@sentry/node` as a dependency:
+<PlatformContent includePath="getting-started-primer" />
+
+This guide explains how to setup Sentry in your Koa application.
+
+Don't already have an account and Sentry project established? Head over to [sentry.io](https://sentry.io/signup/), then return to this page.
+
+<Note>
+This guide is for Version 8.0.0 and up of `@sentry/node`.
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/koa/index.mdx:16]:
```suggestion
This guide is for version 8.0.0 and up of `@sentry/node`.
```

File: docs/platforms/javascript/guides/koa/performance/index.mdx [hunk 10]
```diff
@@ -0,0 +1,24 @@
+---
+title: Performance Monitoring
+description: "Learn more about how to configure our Performance integrations to get the best experience out of it."
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/koa/performance/index.mdx:3]:
```suggestion
description: "Learn more about how to configure our Performance integrations to get the best experience."
```

File: docs/platforms/javascript/guides/koa/performance/index.mdx [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+If you’re adopting Performance in a high-throughput environment, we recommend testing prior to deployment to ensure that your service’s performance characteristics maintain expectations.
+
+</Note>
+
+<PlatformContent includePath="getting-started-config" />
+
+## Automatic Instrumentation
+
+See <PlatformLink to="/performance/instrumentation/automatic-instrumentation">Automatic Instrumentation</PlatformLink> to learn about all the things that the SDK automatically instruments for you.
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/koa/performance/index.mdx:18]:
```suggestion
See <PlatformLink to="/performance/instrumentation/automatic-instrumentation">Automatic Instrumentation</PlatformLink> to learn about everything the SDK automatically instruments for you.
```

File: docs/platforms/javascript/guides/koa/performance/index.mdx [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+## Custom Instrumentation
+
+You can also manually start spans to instrument specific parts of your code. This is useful when you want to measure the performance of a specific operation or function.
+
+See <PlatformLink to="/performance/instrumentation/custom-instrumentation">Custom Instrumentation</PlatformLink> on how to manually start spans.
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/koa/performance/index.mdx:24]:
```suggestion
See <PlatformLink to="/performance/instrumentation/custom-instrumentation">Custom Instrumentation</PlatformLink> to learn how to manually start spans.
```

File: docs/platforms/javascript/guides/node/index.mdx [hunk 13]
```diff
@@ -1,35 +1,50 @@
...
-and learn how to update your code to be compatible with v8.
+This guide explains how to setup Sentry in your Node.js application.
 
 <PlatformContent includePath="alert-using-a-framework" noGuides />
 
 Don't already have an account and Sentry project established? Head over to [sentry.io](https://sentry.io/signup/), then return to this page.
 
-If you prefer to follow video instructions, see [How to Install the Sentry Node SDK in 60 Seconds](https://vimeo.com/899368543).
+<Note>
+This guide is for Version 8.0.0 and up of `@sentry/node`.
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/node/index.mdx:19]:
```suggestion
This guide is for version 8.0.0 and up of `@sentry/node`.
```

File: docs/platforms/javascript/guides/node/performance/index.mdx [hunk 14]
```diff
@@ -0,0 +1,24 @@
+---
+title: Performance Monitoring
+description: "Learn more about how to configure our Performance integrations to get the best experience out of it."
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/node/performance/index.mdx:3]:
```suggestion
description: "Learn more about how to configure our Performance integrations to get the best experience."
```

File: docs/platforms/javascript/guides/node/performance/index.mdx [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+## Automatic Instrumentation
+
+See <PlatformLink to="/performance/instrumentation/automatic-instrumentation">Automatic Instrumentation</PlatformLink> to learn about all the things that the SDK automatically instruments for you.
+
+## Custom Instrumentation
+
+You can also manually start spans to instrument specific parts of your code. This is useful when you want to measure the performance of a specific operation or function.
+
+See <PlatformLink to="/performance/instrumentation/custom-instrumentation">Custom Instrumentation</PlatformLink> on how to manually start spans.
```
Comment (vivianyentran) [inline docs/platforms/javascript/guides/node/performance/index.mdx:24]:
```suggestion
See <PlatformLink to="/performance/instrumentation/custom-instrumentation">Custom Instrumentation</PlatformLink> to learn how to manually start spans.
```

File: includes/node-automatic-instrumentation.mdx [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+- Fastify routes & middlewares
+- HTTP requests made with `http`, `https` and `fetch`
+- DB queries made with:
+  - `mysql`
+  - `mysql2`
+  - `pg`
+  - `graphql` (including Apollo Server)
+  - `mongo`
+  - `mongoose`
+  - `prisma`
```
Comment (vivianyentran) [inline includes/node-automatic-instrumentation.mdx:12]:
```suggestion

```

**Review by vivianyentran — APPROVED** [review]
Added some copyedits to the content changes

**Comment by s1gr1d** [comment]
The docs for serverless are in those commits:
- https://github.com/getsentry/sentry-docs/pull/9906/commits/d353e5de0fe65fc5fd56f9dce29a0eb7629d006b
- https://github.com/getsentry/sentry-docs/pull/9906/commits/16769dae723f236c95a17bb74215240b0600e36c
@mydea

**Comment by mydea** [comment]
> I think the `Configure metrics` section can now be removed from all SDKs since it's automatically configured when you first call the functions.
> 
> For example, there's no need for this section anymore: https://sentry-docs-git-v8-docs.sentry.dev/platforms/javascript/guides/node/metrics/#configure-metrics
> 
> I can submit a separate PR for this?

Yes please, that makes sense!

**Review by HazAT — DISMISSED** [review]
https://github.com/getsentry/sentry-docs/pull/9906/files#diff-b7e7956822e5458c8c74398728fc9e7fdbd160778dc5860580534facf60db755R18-R20
Isn't this wrong?

The error handler always needs to be last, doing a quick local repro confirms this, you need to have all your routes BEFORE you add the error handler

File: docs/platforms/javascript/common/troubleshooting/initialization-methods.mdx [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+```javascript {tabTitle:ESM}
+// Ensure to import this before importing any other modules!
+import "./instrument.js";
+
+// http and other libraries are only instrumented
+// if imported _after_ Sentry has been initialized
+import http from "http";
+```
```
Comment (timfish) [inline docs/platforms/javascript/common/troubleshooting/initialization-methods.mdx:64]: If node is running ESM source, ie. you're not transpiling the above to CJS, my understanding is that import ordering makes zero difference. `import http from "http"` is evaluated and loaded before any code in `instrument.js` is even executed and libraries will not be instrumented without the loader detailed below anyway. 

When you are using the loader, `import-in-the-middle` wraps everything in proxies to work around the whole "imports evaluated first" problem.

Of course, users might be transpiling to CJS, even without their knowledge so maybe it's still important to keep the ordering, but then the loader below will not be required anyway!

**Comment by mydea** [comment]
Updated the PR to show require on top as default installation method, plus added the "Installation Methods" section to node guides showing ESM docs, including when to use which. cc @smeubank

**Review by andreiborza — APPROVED** [review]
LGTM

**Pull request merged**